cpp/compile.sh

```diff
@@ -21,22 +22,22 @@ echo "ARROW_ROOT=${ARROW_ROOT}"
 echo "ARROW_BUILD_FROM_SOURCE_INSTALL_DIR=${ARROW_BFS_INSTALL_DIR}"
 echo "BUILD_JEMALLOC=${BUILD_JEMALLOC}"
 
-CURRENT_DIR=$(cd "$(dirname "$BASH_SOURCE")"; pwd)
-echo $CURRENT_DIR
-cd ${CURRENT_DIR}
-if [ -d build ]; then
+CURRENT_DIR=$(
+  cd "$(dirname "$BASH_SOURCE")"
+  pwd
+)
+echo "${CURRENT_DIR}"
+cd "${CURRENT_DIR}"
+
+if [ "$BUILD_CPP" == "ON" ]; then
+  if [ -d build ]; then
     rm -r build
+  fi
+  mkdir build
+  cd build
+  cmake .. -DTESTS=${TESTS} -DBUILD_ARROW=${BUILD_ARROW} -DSTATIC_ARROW=${STATIC_ARROW} -DBUILD_PROTOBUF=${BUILD_PROTOBUF} -DARROW_ROOT=${ARROW_ROOT} -DARROW_BFS_INSTALL_DIR=${ARROW_BFS_INSTALL_DIR} -DBUILD_JEMALLOC=${BUILD_JEMALLOC}
```

Comment:
Could you make the compilation of libgazelle_cpp.so configurable from Java maven command? If so, we can compile it and put it into jar more conveniently. 
It can be achieved with this commit: https://github.com/rui-mo/gazelle-jni/commit/0f45e2d0cdc3b01eb04fe8020e2c169e4ff9e9b5. If so, we can compile with below command:
` mvn clean package -P full-scala-compiler -DskipTests -Dcheckstyle.skip -Dbuild_protobuf=OFF -Dbuild_gazelle_cpp=ON`

Reply:
Thanks. Will do.